### ISSUE TITLE
Error: Cannot find module 'graphql-request'

### ISSUE THREAD

**Issue by josiahbryan** [weaviate/typescript-client#82]
Error on fresh install of `"weaviate-ts-client": "^1.4.0",`  - it's unusable:
```
/Users/josiahbryan/devel/rubber/backend/node_modules/weaviate-ts-client/dist/index.js:1
Error: Cannot find module 'graphql-request'
Require stack:
- /Users/josiahbryan/devel/rubber/backend/node_modules/weaviate-ts-client/dist/index.js
- /Users/josiahbryan/devel/rubber/backend/src/services/chatbot/utils/graph/node-indexer.js
    at Object.<anonymous> (/Users/josiahbryan/devel/rubber/backend/node_modules/weaviate-ts-client/dist/index.js:1)
    at Generator.next (<anonymous>) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/josiahbryan/devel/rubber/backend/node_modules/weaviate-ts-client/dist/index.js',
    '/Users/josiahbryan/devel/rubber/backend/src/services/chatbot/utils/graph/node-indexer.js'
  ]
}

Node.js v20.5.0
```

**Comment by tsmith023** [comment]
@josiahbryan, I have setup and installed fresh versions of your Node version and the latest client on my Windows and Mac machines without issue! If you are able, a more comprehensive MRE would be helpful to diagnose the problem you're facing.

**Comment by josiahbryan** [comment]
Hmm, will attempt to reproduce again and isolate. Thanks for the quick response! WiIll get back to you in next ~3 hours or so

**Comment by josiahbryan** [comment]
Looks like I needed to manually install deps `graphql` and `graphql-request` - is that expected that I have those installed in **my** project even though I don't use them, but this lib does?

**Comment by tsmith023** [comment]
No, that's quite unexpected since such dependencies should be private and are not peer dependencies! I only ran `npm install weaviate-ts-client` in my empty `package.json` file without issues 🤔 

My guess is that you may have some machine-specific global configuration of `npm` specified in your `.npmrc` file that is affecting the installation process. If this is not the case, then we can troubleshoot otherwise if you replicate the installation process with verbose logging:
1. Create a fresh `package.json` file by running `npm init` in an empty directory
2. Run `npm install --verbose weaviate-ts-client`
3. Inspect the terminal logs for any installation errors, pasting the log output here

**Comment by zoranbabic** [comment]
I ran into a similar issue. We are using `pnpm` in our project, which doesn't install peer dependencies automatically.
The `weaviate` SDK package has a dependency to `graphql-request` which has a peer dependency to `graphql`. After installing `weaviate-ts-client` I received this warning:

```
WARN  Issues with peer dependencies found
packages/search
└─┬ weaviate-ts-client 1.5.0
  └─┬ graphql-request 5.2.0
    ├── ✕ missing peer graphql@"14 - 16"
    └─┬ @graphql-typed-document-node/core 3.2.0
      └── ✕ missing peer graphql@"^0.8.0 || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0"
Peer dependencies that should be installed:
  graphql@">=14.0.0 <15.0.0 || >=15.0.0 <16.0.0 || >=16.0.0 <17.0.0"  

```

The SDK of course wouldn't work and complain: `Error: Cannot find module 'graphql'` - until I installed `graphql` manually.

**Comment by malgamves** [comment]
I use `yarn` and had the same issue, installing graphql fixed it

> The SDK of course wouldn't work and complain: Error: Cannot find module 'graphql' - until I installed graphql manually.

**Comment by geminigeek** [comment]
same issue version 2.0.0

**Comment by j-d-carmichael** [comment]
Not quite the same but in the same area:  https://github.com/weaviate/typescript-client/issues/106

**Comment by g-despot** [comment]
This was against `weaviate-ts-client` v1, which has since been replaced by `weaviate-client` v3.
Closing as obsolete. If you hit an install problem on v3, please open a fresh issue, thanks to everyone who added their setup here, it helped.
